lib/puma/reactor.rb

```diff
@@ -16,17 +17,44 @@ def initialize(server, app_pool)
       @sleep_for = DefaultSleepFor
       @timeouts = []
 
-      @sockets = [@ready]
+      @selector = NIO::Selector.new
+      @selector.register(@ready, :r).value = nil
```

Comment:
You can replace this with the `NIO::Selector#wakeup` mechanism, which is basically doing the same thing behind the scenes